contracts/modules/Offers/Omnibus/OffersOmnibus.sol

```diff
@@ -0,0 +1,350 @@
+
+        if (_offerCurrency != address(0)) {
+            require(msg.value == 0, "MSG_VALUE_GT_ZERO_WITH_OTHER_CURRENCY");
+            IERC20 token = IERC20(_offerCurrency);
+            require(token.balanceOf(msg.sender) >= _offerAmount, "INSUFFICIENT_BALANCE");
+            require(token.allowance(msg.sender, address(erc20TransferHelper)) >= _offerAmount, "INSUFFICIENT_ALLOWANCE");
+        } else {
+            require(msg.value == _offerAmount, "MSG_VALUE_NEQ_OFFER_AMOUNT");
+            weth.deposit{value: msg.value}();
+            weth.transferFrom(address(this), msg.sender, _offerAmount);
+            require(weth.balanceOf(msg.sender) >= _offerAmount, "INSUFFICIENT_BALANCE");
+            require(weth.allowance(msg.sender, address(erc20TransferHelper)) >= _offerAmount, "INSUFFICIENT_ALLOWANCE");
+        }
+        require(erc721TransferHelper.isModuleApproved(msg.sender), "MODULE_NOT_APPROVED");
+
+        offer.maker = msg.sender;
+        offer.amount = _offerAmount;
+        offer.features = 0;
+
+        _setETHorERC20Currency(offer, _offerCurrency);
+
+        if (_listingFee.listingFeeBps > 0) {
+            require(_listingFee.listingFeeBps <= 10000, "INVALID_LISTING_FEE");
+            _setListingFee(offer, _listingFee.listingFeeBps, _listingFee.listingFeeRecipient);
+        }
+
+        if (_findersFeeBps > 0) {
+            require(_findersFeeBps <= 10000, "createOffer finders fee bps must be less than or equal to 10000");
+            require(_findersFeeBps + _listingFee.listingFeeBps <= 10000, "listingFee and findersFee must be less than or equal to 10000");
+            _setFindersFee(offer, _findersFeeBps);
+        }
+
+        if (_expiry > 0) {
+            require(_expiry > block.timestamp, "Expiry must be in the future");
+            _setExpiry(offer, _expiry);
+        }
+
+        emit OfferCreated(_tokenContract, _tokenId, offerCount, _getFullOffer(offer));
+        return offerCount;
+    }
+
+    function _validateAndPrepareNewOffer(
+        address _oldCurrency,
+        address _newCurrency,
+        uint256 _oldAmount,
+        uint256 _newAmount
+    ) internal {
+        require(_newAmount > 0, "NO_ZERO_OFFERS");
+        require(_newAmount != _oldAmount || _newCurrency != _oldCurrency, "SAME_OFFER");
+
+        IERC20 token;
+
+        if (_newCurrency == address(0)) {
+            if (_oldCurrency == address(0)) {
+                if (_newAmount > _oldAmount) {
+                    require(msg.value == _newAmount - _oldAmount, "INVALID_MSG_VALUE");
+                } else {
+                    require(msg.value == 0, "INVALID_MSG_VALUE");
+                }
+            } else {
+                require(msg.value == _newAmount, "INVALID_MSG_VALUE");
+            }
+            token = IERC20(address(weth));
+        } else {
+            require(msg.value == 0, "INVALID_MSG_VALUE");
+            token = IERC20(_newCurrency);
+        }
+
+        if (msg.value > 0) {
```

Comment:
What if you just insert the WETH before finalizing the token?

The logic here seems a bit strange: `require(msg.value == _newAmount - _oldAmount, "INVALID_MSG_VALUE");`

Nested IFs usually aren't great: I wonder if you can safely convert any value to WETH and set the weth token address before hitting this function then it just validates that the WETH value is enough. No need to directly validate the WETH value or use msg.value with this approach and if you don't have enough balance no matter how you set it up the function fails. You could also accept deposits to top up WETH and use the value the user passes in to set the final amount that way it's up to the user and the UI to handle this.